### ISSUE TITLE
Core/GameObject: Gob interraction is not updated when the player take a quest

### ISSUE THREAD

**Issue by Killyana** [TrinityCore/TrinityCore#24952]
**Description:** When a player take a quest that turns some gobs clickable + sparkles, the state of this gobs will not update until they are reloaded in the client.

**Steps to reproduce the problem:**

1. Connect a horde  char
2. Don't use (gm on/off) during the test as it will update visibility
2. .quest add 894
3. .go game 13167
4. complete the quest and take the next one
5. notice the 3 valves cannot be clicked and they don't sparkle
6. move a way and come back and the issue will be fixed or just use .gm off

Could be the result of https://github.com/TrinityCore/TrinityCore/commit/449bd9058bce87ade6831ba38d1a0b83b9cbfb18

**Branch(es):** 3.3.5
**TC rev. hash/commit:**  rev. cc71da35b5dc

**Comment by Shauren** [comment]
449bd9058bce87ade6831ba38d1a0b83b9cbfb18 has nothing to do with this one, valves are not lootable for quest items (that commit only changes behavior after you loot/destroy item, not on quest accept)

**Comment by dr-j** [comment]
yes was going to say with chests this works fine, to test start quest wandering shay or the sleeper has awakened once you have taken quest no need to actualy do quest but examine the chest to loot the quest item to wake npc or get attention.

**Comment by TQPS** [comment]
Came across this issue today with [Samophlange](https://tbc.wowhead.com/quest=900/samophlange#comments) quest.

Workaround in the meantime - log out and back in seemed to enable the objects.

**Comment by Legoso** [comment]
TrinityCore rev. 0c1e5fea69bb+ 2024-01-04 20:55:19 +0100 (3.3.5 branch) (Win64, Release, Static)

**TDB 335.23061** and updates up to TrinityCore rev. 0c1e5fea69bb .

Confirmed. I used to think it was caused by my game client addons, but I tested it now without any addons loaded, just to make sure.

**Samophlange quest chain, minimum level 10, recommended level 14-16:**
- https://aowow.trinitycore.info/?quest=894 (starting quest picked up from [Sputtervalve](https://aowow.trinitycore.info/?npc=3442) in Ratchet, between the flight master and the bank)
- **https://aowow.trinitycore.info/?quest=900 (valve gameobjects activation issue)**
- https://aowow.trinitycore.info/?quest=901 (console key retrieval)
- https://aowow.trinitycore.info/?quest=902 (return to Ratchet)

Short description:
- Upon taking quest 900, the valves should turn active, but they do not. You have to log out and log in again to activate the valves. It is not enough to abandon the quest and retake it from the console.
- If you abandon quest 900 after logging in to see the gameobjects active, the valves stay active, so you have to log out and in again to turn them off.


Notable objects:
- https://aowow.trinitycore.info/?object=61936/fuel-control-valve
- https://aowow.trinitycore.info/?object=61935/regulator-valve
- https://aowow.trinitycore.info/?object=4072/main-control-valve
- https://www.wowhead.com/wotlk/object=4141/control-console

**Comment by Maiievs** [comment]
The main logic for this main behavior is implemented in UpdateVisibleGameobjectsOrSpellClicks,
but it doesn't work for the quest Samophlange for instance because the rule is set to only consider objects with type goobers that have Data1 > 0, and for these objects Data1 is -1.

For objects of type Goober there is 90 cases that have Data1 <=0 and are listed as required for quest objectives.
